### PR TITLE
Molten Core fixes

### LINKED ISSUE #595: [RAID] MC Ragnaros' hammer falls facing the wrong direction.
**Description**: The hammer is falling so the side of the hammer is not visible to the player after Raganros is killed. Having farmed this countless times in retail, and claimed the Eye of Sulfuras myself, I know this is wrong.

**Expected behavior:** 
When the hammer falls, it should land in the lava with the side facing the players giving a full view of the epic hammer from the fight start position. This is the same direction Ragnaros is facing when the fight starts. A bit hard to describe, but if you go check the image you'll see what I mean. 

Right: https://www.flickr.com/photos/erocsid/4021735116/in/album-72157622608658698/
Wrong: https://imgur.com/a/Xi2t8

**Branch:**
Master

**Commit Hash:** 
c15f15ebb0a113a5cf20fc11afb89b2bcff9ec6d

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/48908411-raid-mc-ragnaros-hammer-falls-facing-the-wrong-direction?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by Gargarensis**
**Changes proposed:**

-  Fix Ragnaros' death orientation
-  Runes in Molten Core now extinguish when their boss dies
-  Golemagg and Majordomo's adds now will always join the fight
-  Core Hound script
-  Majordomo and adds position adjusted

**Target branch(es):** master

**Issues addressed:** Closes #595 #593 #592 #598 #597 #594

**Tests performed:** Compiled on windows, it build, it seems to work, but further tests might be very helpful.

**Known issues and TODO list:**

- [Core Hound's Serrated Bites timer might be wrong (10 seconds)] 
- [Core Hound's smoldering animation is not the original one since I could not find the right spell]

### CODE REVIEW

File: src/scripts/EasternKingdoms/BlackrockMountain/MoltenCore/boss_golemagg.cpp [hunk 1]
```diff
@@ -61,7 +61,18 @@ class boss_golemagg : public CreatureScript
...
+				std::list<Creature *> ragers;
+				me->GetCreaturesWithEntryInRange(ragers, 100, NPC_CORE_RAGER);
+				for (Creature * i : ragers)
+				{
+					if (i && i->IsAlive() && !i->IsInCombat())
+					{
+						i->AI()->AttackStart(victim);
+					}
+				}
+			}
```
Comment (ShinDarth) [inline src/scripts/EasternKingdoms/BlackrockMountain/MoltenCore/boss_golemagg.cpp]: please fix indentation

File: src/scripts/EasternKingdoms/BlackrockMountain/MoltenCore/boss_magmadar.cpp [hunk 2]
```diff
@@ -18,22 +18,27 @@ EndScriptData */
 
 enum Texts
 {
-    EMOTE_FRENZY        = 0
+    EMOTE_FRENZY        = 0,
+	EMOTE_SMOLDERING    = 0,
+	EMOTE_IGNITE		= 1,
```
Comment (ShinDarth) [inline src/scripts/EasternKingdoms/BlackrockMountain/MoltenCore/boss_magmadar.cpp]: please fix indentation

File: src/scripts/EasternKingdoms/BlackrockMountain/MoltenCore/boss_magmadar.cpp [hunk 3]
```diff
@@ -18,22 +18,27 @@ EndScriptData */
...
 {
-    SPELL_FRENZY        = 19451,
-    SPELL_MAGMA_SPIT    = 19449,
-    SPELL_PANIC         = 19408,
-    SPELL_LAVA_BOMB     = 19428,
+	SPELL_FRENZY			= 19451,
+	SPELL_MAGMA_SPIT		= 19449,
+	SPELL_PANIC				= 19408,
+	SPELL_LAVA_BOMB			= 19428,
+	SPELL_SERRATED_BITE		= 19771,
```
Comment (ShinDarth) [inline src/scripts/EasternKingdoms/BlackrockMountain/MoltenCore/boss_magmadar.cpp]: please fix indentation

File: src/scripts/EasternKingdoms/BlackrockMountain/MoltenCore/boss_magmadar.cpp [hunk 4]
```diff
@@ -18,22 +18,27 @@ EndScriptData */
...
 };
 
 enum Events
 {
     EVENT_FRENZY        = 1,
     EVENT_PANIC         = 2,
     EVENT_LAVA_BOMB     = 3,
+	EVENT_SERRATED_BITE = 1,
+	EVENT_IGNITE		= 2,
```
Comment (ShinDarth) [inline src/scripts/EasternKingdoms/BlackrockMountain/MoltenCore/boss_magmadar.cpp]: please fix indentation

File: src/scripts/EasternKingdoms/BlackrockMountain/MoltenCore/boss_magmadar.cpp [hunk 5]
```diff
@@ -104,7 +109,103 @@ class boss_magmadar : public CreatureScript
...
+	CreatureAI* GetAI(Creature* creature) const
+	{
+		return new npc_magmadar_core_houndAI(creature);
+	}
+};
+
 void AddSC_boss_magmadar()
 {
     new boss_magmadar();
+	new npc_magmadar_core_hound();
```
Comment (ShinDarth) [inline src/scripts/EasternKingdoms/BlackrockMountain/MoltenCore/boss_magmadar.cpp]: please fix indentation

File: src/scripts/EasternKingdoms/BlackrockMountain/MoltenCore/boss_majordomo_executus.cpp [hunk 6]
```diff
@@ -80,6 +80,8 @@ class boss_majordomo : public CreatureScript
                 events.ScheduleEvent(EVENT_DAMAGE_REFLECTION, 15000);
                 events.ScheduleEvent(EVENT_BLAST_WAVE, 10000);
                 events.ScheduleEvent(EVENT_TELEPORT, 20000);
+				// Call every flamewaker around him
+				me->CallForHelp(30);
```
Comment (ShinDarth) [inline src/scripts/EasternKingdoms/BlackrockMountain/MoltenCore/boss_majordomo_executus.cpp]: please fix indentation

File: src/scripts/EasternKingdoms/BlackrockMountain/MoltenCore/boss_ragnaros.cpp [hunk 7]
```diff
@@ -83,6 +84,7 @@ class boss_ragnaros : public CreatureScript
                 _hasSubmergedOnce = false;
                 _isBanished = false;
                 me->SetUInt32Value(UNIT_NPC_EMOTESTATE, 0);
+				me->SetOrientation(0.8f);
```
Comment (ShinDarth) [inline src/scripts/EasternKingdoms/BlackrockMountain/MoltenCore/boss_ragnaros.cpp]: please fix indentation

File: src/scripts/EasternKingdoms/BlackrockMountain/MoltenCore/instance_molten_core.cpp [hunk 8]
```diff
@@ -89,6 +89,27 @@ class instance_molten_core : public InstanceMapScript
...
+						break;
+					case GO_CIRCLE_MAGMADAR:
+						_circlesGUIDs[1] = go->GetGUID();
+						break;
+					case GO_CIRCLE_SHAZZRAH:
+						_circlesGUIDs[4] = go->GetGUID();
+						break;
+					case GO_CIRCLE_SULFURON:
+						_circlesGUIDs[6] = go->GetGUID();
+						break;
```
Comment (ShinDarth) [inline src/scripts/EasternKingdoms/BlackrockMountain/MoltenCore/instance_molten_core.cpp]: please fix indentation

File: src/scripts/EasternKingdoms/BlackrockMountain/MoltenCore/instance_molten_core.cpp [hunk 9]
```diff
@@ -137,6 +158,20 @@ class instance_molten_core : public InstanceMapScript
...
+						circle->Delete();
+						circle->SetRespawnTime(7 * DAY);
+						_circlesGUIDs[bossId] = 0;
+				}
+					
+				if (state == DONE && bossId == BOSS_MAGMADAR)
+				{
+
+				}
+
```
Comment (ShinDarth) [inline src/scripts/EasternKingdoms/BlackrockMountain/MoltenCore/instance_molten_core.cpp]: please fix indentation

File: src/scripts/EasternKingdoms/BlackrockMountain/MoltenCore/instance_molten_core.cpp [hunk 10]
```diff
@@ -238,6 +273,7 @@ class instance_molten_core : public InstanceMapScript
             uint8 _ragnarosAddDeaths;
             bool _isLoading;
             bool _summonedExecutus;
+			std::unordered_map<uint8, uint64> _circlesGUIDs;
```
Comment (ShinDarth) [inline src/scripts/EasternKingdoms/BlackrockMountain/MoltenCore/instance_molten_core.cpp]: please fix indentation

File: src/scripts/EasternKingdoms/BlackrockMountain/MoltenCore/molten_core.h [hunk 11]
```diff
@@ -47,11 +47,20 @@ enum Creatures
...
 enum GameObjects
 {
     GO_CACHE_OF_THE_FIRELORD        = 179703,
+	GO_CIRCLE_SULFURON				= 178187,
+	GO_CIRCLE_BARON					= 178188,
+	GO_CIRCLE_SHAZZRAH				= 178189,
+	GO_CIRCLE_GOLEMAGG				= 178190,
+	GO_CIRCLE_GARR					= 178191,
+	GO_CIRCLE_MAGMADAR				= 178192,
+	GO_CIRCLE_GEHENNAS				= 178193,
```
Comment (ShinDarth) [inline src/scripts/EasternKingdoms/BlackrockMountain/MoltenCore/molten_core.h]: please fix indentation

File: src/scripts/EasternKingdoms/BlackrockMountain/MoltenCore/molten_core.h [hunk 12]
```diff
@@ -47,11 +47,20 @@ enum Creatures
     NPC_RAGNAROS                    = 11502,
     NPC_FLAMEWAKER_HEALER           = 11663,
     NPC_FLAMEWAKER_ELITE            = 11664,
+	NPC_CORE_RAGER					= 11672,
+	NPC_CORE_HOUND					= 11671,
```
Comment (ShinDarth) [inline src/scripts/EasternKingdoms/BlackrockMountain/MoltenCore/molten_core.h]: please fix indentation

**Comment by Yehonal** [comment]
Please merge updates from master branch, we need to run a travis build to check about warnings.

File: src/scripts/EasternKingdoms/BlackrockMountain/MoltenCore/boss_majordomo_executus.cpp (hunk 6 above)
Comment (StygianTheBest) [reply src/scripts/EasternKingdoms/BlackrockMountain/MoltenCore/boss_majordomo_executus.cpp]: I am still able to kill the healers without the rest attacking me.

File: src/scripts/EasternKingdoms/BlackrockMountain/MoltenCore/boss_ragnaros.cpp (hunk 7 above)
Comment (StygianTheBest) [reply src/scripts/EasternKingdoms/BlackrockMountain/MoltenCore/boss_ragnaros.cpp]: The hammer is still not facing the right way, but it appears to have moved in the right direction from where it used to fall.

File: src/scripts/EasternKingdoms/BlackrockMountain/MoltenCore/boss_magmadar.cpp (hunk 5 above)
Comment (StygianTheBest) [reply src/scripts/EasternKingdoms/BlackrockMountain/MoltenCore/boss_magmadar.cpp]: Just saw your comments at the top here after I posted on the issue.

File: src/scripts/EasternKingdoms/BlackrockMountain/MoltenCore/boss_ragnaros.cpp (hunk 7 above)
Comment (StygianTheBest) [reply src/scripts/EasternKingdoms/BlackrockMountain/MoltenCore/boss_ragnaros.cpp]: There is now a LARGE glowing orb of fire and some blood spatter falling down above the hammer which should not be there.

**Comment by StygianTheBest** [comment]
- The hammer is still not falling in the correct position, and now there's a large glowing orb of fire with some blood splatter raining down above the hammer that should not be there.
- The Flamewalker Healers with Majordomo can still be killed without the others attacking. I killed two with heroic throw and nothing aggro'd.
- I've listed the issues with the Core Hounds on the other issue entry, but I saw the comments at the top of the commit that you're aware of that after I posted.

**Comment by Gargarensis** [comment]
I don't know where that orb comes from, I just change the orientation at the moment Ragnaros' dies, maybe the value I've set is wrong?
I will do something else for the Healers if the current solution does not work, do you 1-shot them with your ability?

**Comment by StygianTheBest** [comment]
@Gargarensis 

- That's a great point. Yes, I was one-shotting them. I'll re-test and see what happens.

**Comment by StygianTheBest** [comment]
@Gargarensis I saved over 10,000 screenshots from my days of WoW, and I unarchived them and confirmed that the "large glowing orb of fire with some blood splatter raining down" is actually Blizzlike and supposed to be there. This screenshot was dated 2010.

**Comment by Jagermaister95** [comment]
I tested the fixes made and I can say that:
1) The bat when dying ragnaros is smaller and in the wrong direction (it should be so https://vignette.wikia.nocookie.net/wowwiki/images/8/86/World_of_Warcraft_Game-Wide_First_Ragnaros_Kill/revision/latest/scale-to- width-down / 425? cb = 20130618030202).
2) The Flamewalker Healers with Majordomo they do not die with just one spell and when you hit them the fight begins correctly.
3)I've been searching for numerous videos and pictures and large glowing orb of fire with some blood splatter raining down not appeared when Ragnaros dies.
4)When you kill a Core Hound you see the animation but it recovers all your life and stays as if it was dead.

**Comment by StygianTheBest** [comment]
#595 and #592 are not fully resolved.

**Pull request merged**